todoList/test/todoList.test.js

```diff
@@ -0,0 +1,354 @@
+    input.value = 'completed';
+    simulateEnterKeyPress(input);
+    simulateClick(document.querySelector('.todo-checkbox'));
+
+    input.value = 'acive 1';
+    simulateEnterKeyPress(input);
+
+    input.value = 'active 2';
+    simulateEnterKeyPress(input);
+
+    input.value = 'to complete';
+    simulateEnterKeyPress(input);
+
+    simulateClick(document.querySelector('.todo-checkbox'));
+
+    expect(document.querySelectorAll('.todos')[3].textContent).toBe('to complete');
+  });
+
+  it('should move to the top of todo list if toggle active', () => {
+    let toAciveChekbox;
+
+    input.value = 'completed';
+    simulateEnterKeyPress(input);
+    simulateClick(document.querySelector('.todo-checkbox'));
+
+    input.value = 'to active';
+    simulateEnterKeyPress(input);
+    toAciveChekbox = document.querySelector('.todo-checkbox');
+    simulateClick(toAciveChekbox);
+
+    input.value = 'acive 1';
+    simulateEnterKeyPress(input);
+
+    input.value = 'active 2';
+    simulateEnterKeyPress(input);
+
+    simulateClick(toAciveChekbox);
+
+    expect(document.querySelectorAll('.todos')[0].textContent).toBe('to active');
+  });
+
+  it('should set other class to completed todos for apply style different from active todos', () => {
+    input.value = 'completed';
+    simulateEnterKeyPress(input);
+    simulateClick(document.querySelector('.todo-checkbox'));
+
+    input.value = 'acive';
+    simulateEnterKeyPress(input);
+
+    expect(document.querySelectorAll('.complete').length).toBe(1);
+  });
+});
+
+describe('Test to count todos in incomplete', () => {
+  let input;
+  beforeEach(() => {
+    const todoList = new TodoList(document.querySelector('#target'));
+    input = document.querySelector('.todo-input');
+  });
+
+  it('should zero when initialized', () => {
+    expect(document.querySelector('.todo-remain').textContent).toBe('0');
+  });
+
+  it('should have a number of incomplete todos.', () => {
+    input.value = 'todo';
+    simulateEnterKeyPress(input);
+
+    expect(document.querySelector('.todo-remain').textContent).toBe('1');
+  });
+
+  it('should zero if all todos are completed', () => {
+    input.value = 'todo';
+    simulateEnterKeyPress(input);
+    simulateClick(document.querySelector('.todo-checkbox'));
+
+    expect(document.querySelector('.todo-remain').textContent).toBe('0');
+  });
+});
+
+describe('Test to action of view filter buttons', () => {
+  let input;
+  beforeEach(() => {
+    const todoList = new TodoList(document.querySelector('#target'));
+    input = document.querySelector('.todo-input');
+  });
+
+  it('should set all button to selected button when initialize', () => {
+    expect(document.querySelector('.selected-btn').name).toBe('all');
+  });
+
+  it('should set acive button to selected button when acive button clicked', () => {
+    simulateClick(document.querySelector('.todo-active-btn'));
+
+    expect(document.querySelector('.selected-btn').name).toBe('active');
```

Comment:
이런 패턴의 matcher가 반복된다면... 테스트용 유틸 함수를 추가하는것도 좋습니다. ㅎㅎ

```suggestion
    getPropValue(selector, propName) {
        const el = document.querySelector(selector);
        return el ? el[propName] : null;
    }

        expect(getPropVaule('.selected-btn', 'name').toBe('active');
```

Reply:
이 기능 아주 좋군요..

Reply:
@jung-han `cmd + g` 누르면 `suggestion` 언어 타입으로 코드블럭이 자동 생성되는데요... 아주 좋습니다 ㅎㅎ

단, 댓글 달 때 복수 라인을 비교하려면 드래그해서 생성해야했던 것 같아요 ㅎㅎ

Reply:
위 경우를 포함해 반복되는 패턴의 matcher에 해당하는 데스트용 유틸 함수를 만들어 사용했습니다.